### DISCUSSION TITLE
Seamless Playback?

### DISCUSSION THREAD

**Discussion by mmattbtw** [rekkyrosso/ampcast#26] [Ideas]
Right now, when trying to listen to DJ Mixes from Apple Music, or DJ Mixes I have saved on my Jellyfin servers. These mixes are split up by the tracks played, but services like Apple Music offer a seamless playback mode so that the tracks go right into each other without any stopping, delay, or anything.

At the moment, ampcast doesn't do this which makes it kinda annoying to try and listen to these mixes. This is likely a much harder problem then it should be so I wish good luck all around 🙏

**Comment by rekkyrosso** [comment]
Gapless playback is on my list of things to do. But as you guessed it is quite tricky.

**Comment by rekkyrosso** [comment]
This was just recently implemented. Version 0.9.14.

**Comment by vinyln0va** [comment]
Hi,

just to report not experiencing any gapless playback has this been implemented? would also love to see a crossfade feature added in future releases if feasible..

**Reply by rekkyrosso** [reply]
It's only gapless if the actual audio tracks have no opening/trailing silence.

Eliminating silence requires some audio analysis. That's pretty hard using web audio but can be implemented using a binary player in the desktop app. That would also provide crossfading. It's on the roadmap but nothing is planned for the immediate future.
